In [1]:
import nbimporter

In [2]:
import numpy as np

In [3]:
from math import factorial, e

In [4]:
from Formula import *

#### 1 - Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

Находится по формуле Бернулли: $ P_{n} = C_{n}^{k}p^{k}q^{n-k} $<br>
где: 
- n = 100<br>
- k = 85<br>
- p = 0.8<br>
- q = 1 - p

In [6]:
bernulli(85, 100, 0.8)

4.806179370074635

Ответ: 4.8%

#### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

Находится по формуле Пуассона: $P_{m} = \frac{\lambda^{m}} {m!}e^{-\lambda} $<br>
Какова вероятность, что ни одна из них не перегорит в первый день?<br>
где:
- n = 5000<br>
- m = 0<br>
- p = 0.004<br>
- $\lambda $ = P * n<b>
- e $\approx$ 2.72    
Какова вероятность, что перегорят ровно две?<br>
где: 
- n = 5000<br>
- m = 2<br>
- p = 0.004<br>
- $\lambda $ = P * n<br>
- e $\approx$ 2.72

In [7]:
puasson(0, 5000, 0.0004)

13.53352832366127

In [8]:
puasson(2, 5000, 0.0004)

27.06705664732254

Ответ: 
- ни одна из них не перегорит $ \approx $ 13.5%
- перегорят ровно две $ \approx $ 27%

#### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

Находится по формуле Бернулли: $ P_{n} = C_{n}^{k}p^{k}q^{n-k} $<br>
где: 
- n = 144<br>
- k = 70<br>
- p = 0.5<br>
- q = 1 - p

In [9]:
bernulli(70, 144, 0.5)

6.281178035144776

Ответ: 6.2%

#### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

- #### Какова вероятность того, что все мячи белые?

$ \frac{C_{7}^{2}}{C_{10}^{2}} * \frac{C_{9}^{2}}{C_{11}^{2}} $

In [8]:
((soch(2, 7)/soch(2, 10))*(soch(2, 9)/soch(2, 11)))*100

30.545454545454547

Ответ: 30.5%

- #### Какова вероятность того, что ровно два мяча белые?

$ \frac{С^{2}_{7}}{С^{2}_{10}} * \frac{С^{2}_{2}}{С^{2}_{11}} +  \frac{С^{2}_{3}}{С^{2}_{10}} * \frac{С^{2}_{9}}{С^{2}_{11}} + \frac{С^{1}_{7}С^{1}_{3}}{C^{2}_{10}} * \frac{С^{1}_{9}С^{1}_{2}}{C^{2}_{11}} $

In [11]:
((soch(2, 7)/soch(2, 10))*(soch(2, 3)/soch(2, 11))+(soch(2, 3)/soch(2,10))*(soch(2, 9)/soch(2, 11))+\
((soch(1, 7)*soch(1, 3))/soch(2, 10)) * ((soch(1, 9)*soch(1, 2))/soch(2, 11)))*100

22.181818181818183

Ответ: 22.18%

- #### Какова вероятность того, что хотя бы один мяч белый?

Данная вероятность обратная вероятности того, что все мячи будут черные.<br>
$ 1 -\frac{С^{2}_{3}}{С^{2}_{10}} * \frac{С^{2}_{2}}{С^{2}_{11}} $

In [13]:
1 - ((soch(2, 3)/soch(2, 10))*(soch(2, 2)/soch(2, 11)))

0.9987878787878788

Ответ: 0.9987%